packages/server/tests/integration/server.spec.ts

```diff
@@ -423,6 +423,215 @@ describe('RPC Server', async function () {
     expect(res.data.result).to.be.equal('0x0');
   });
 
+  describe('batchRequest Test Cases', async function () {
+    const batchRequestEnabledValue = process.env.BATCH_REQUESTS_ENABLED;
+
+    this.beforeAll(function () {
+      process.env.BATCH_REQUESTS_ENABLED = 'true';
+    });
+
+    this.afterAll(function () {
+      process.env.BATCH_REQUESTS_ENABLED = batchRequestEnabledValue;
+    });
+
+    function getEthChainIdRequest(id) {
+      return {
+        id: `${id}`,
+        jsonrpc: '2.0',
+        method: RelayCalls.ETH_ENDPOINTS.ETH_CHAIN_ID,
+        params: [null],
+      };
+    }
+
+    function getEthAccountsRequest(id) {
+      if (id == null) {
+        return {
+          jsonrpc: '2.0',
+          method: RelayCalls.ETH_ENDPOINTS.ETH_ACCOUNTS,
+          params: [null],
+        };
+      } else {
+        return {
+          id: `${id}`,
+          jsonrpc: '2.0',
+          method: RelayCalls.ETH_ENDPOINTS.ETH_ACCOUNTS,
+          params: [null],
+        };
+      }
+    }
+
+    function getNonExistingMethodRequest(id) {
+      return {
+        id: `${id}`,
+        jsonrpc: '2.0',
+        method: 'non_existent_method',
+        params: [null],
+      };
+    }
+
+    it('should execute "eth_chainId" in batch request', async function () {
+      // 3 request of eth_chainId
+      const response = await this.testClient.post('/', [
+        getEthChainIdRequest(2),
+        getEthChainIdRequest(3),
+        getEthChainIdRequest(4),
+      ]);
+
+      // verify response
+      BaseTest.baseDefaultResponseChecks(response);
+
+      // verify response for each request
+      for (let i = 0; i < response.data.length; i++) {
+        expect(response.data[i].id).to.be.equal((i + 2).toString());
+        expect(response.data[i].result).to.be.equal('0x' + Number(process.env.CHAIN_ID).toString(16));
+      }
+    });
+
+    it('should execute "eth_chainId" and "eth_accounts" in batch request', async function () {
```

Comment:
Would be great to have an acceptance test and add it to `@release` that covers the standard flow of

1. chain id
2. account nonce
3. gas estimate
4. submit raw transaction
5. transaction receipt
6. contract state call

Reply:
just created a new issue for follow up on this 👍